### ISSUE TITLE
Feature Request / Consistency: Easier way to switch to music mode

### ISSUE THREAD

**Issue by whatsnewsisyphus** [iina/iina#2339]
Hello
1.0.2 b105
OSX 10.13.16

It is very easy to switch out of the Music mode / accidentally close it's window but it's hard to switch into music mode. Perhaps it would be useful to add a switch back to music mode somewhere? The music mode still looks a bit like a draft in how different it is (non-native window and so forth) from the rest of the app. I would also move the 'music mode' from the Video menu to Audio or Playback if View/Window are going to be untouched.

Thanks a million for IINA

**Comment by LEoREo2247** [comment]
Actually, you can do that by adding toolbar item from preferences.
First, go to preferences, then UI. In "On Screen Controller" section, you will find Customize button of toolbar.
Click that and drag the "Enter music mode" item to current items, and now you can switch to music mode any time from the controller.
It's kinda hard to find but thats what IINA has for now.

**Comment by anohren** [comment]
Perhaps it would be a good idea to expose this feature more, e.g. by following the macOS convention of right clicking the toolbar to customize it.

**Comment by LEoREo2247** [comment]
> right clicking the toolbar to customize it.

That may be a intuitive solution. We could leave the current customize button in preferences but just add the right click customize menu on toolbar.

**Comment by anohren** [comment]
That was my thought as well. Just a menu in the toolbar that opens the right tab in the settings window.

**Comment by LEoREo2247** [comment]
> right clicking the toolbar to customize it.

That may be a intuitive solution. We could leave the current customize button in preferences but just add the right click customize menu on toolbar.

I'll talk more about consistency later, I just have to go for now.

**Comment by whatsnewsisyphus** [comment]
If the music mode works by default wouldn't it also make sense to show the switch to music mode icon in the toolbar by default as well, at least for traditionally audio only filettypes?

**Comment by anohren** [comment]
> If the music mode works by default

Not sure what you meant here. That it gets activated automatically by default?

> show the switch to music mode icon in the toolbar by default as well, at least for traditionally audio only filettypes

I was under the impression that the existing option to automatically switch to music mode when playing an audio file was enabled by default, which would lessen the need for this. I might be mistaken.

So basically your suggestion is to automatically add the music mode toolbar item whenever an audio file is playing in video mode. That could potentially be an option in the toolbar customization panel: "Only show this item while playing audio files"

Automatically customizing the toolbar based on file type starts to blur the line between the two modes though, which begs the question why not just enable the existing option to automatically enter music mode based on file type. If the reason is that there's something that doesn't work as well in music mode, then that issue should probably get some attention.

> (non-native window and so forth) 

In what way is the window non-native?

**Comment by whatsnewsisyphus** [comment]
Hell
> 
> I was under the impression that the existing option to automatically switch to music mode when playing an audio file was enabled by default, which would lessen the need for this. I might be mistaken.
Yes it does switch to music mode, but if you switch back from music mode, there is no corresponding UI button to go back.
> 
> In what way is the window non-native?
Instead of using the OS window controls it uses a little X and a 'Back to Video Mode' button visible on album art on hover, which is fine, but the video window doesn't have that hover UI, so it feels like it is another app or something.

**Comment by LEoREo2247** [comment]
I agree on the point that the window design between the music mode and the normal window is somewhat inconsistent.
At least I'm not a fan of the control button below the X button to exit the music mode.

About the menu item, I just simply thought that if the PiP item is in Video menu, then the music mode item could also fit there well. But I also agree that the current Window menu is not utilized well.

**Comment by anohren** [comment]
> Yes it does switch to music mode, but if you switch back from music mode, there is no corresponding UI button to go back.

Got it. 

Removing the quick switch button would technically solve the problem, but would be very detrimental to those who accidentally get stuck playing a video in music mode unlike the other way around. 

Since the display area isn't really used for anything critical while playing music, how about introducing a button on it on mouseover when playing audio? It's an alternative at least.

> Instead of using the OS window controls it uses a little X and a 'Back to Video Mode' button visible on album art on hover, which is fine, but the video window doesn't have that hover UI, so it feels like it is another app or something.

I also have some issues concerning those buttons (#2127) so I hear you, but I recognize that they're a pretty clever compromise with some thought put into it. So, perhaps a "Enter music mode (→)" button in the same style when hovering over the album art in video mode would visually tie the two together somewhat.

**Comment by whatsnewsisyphus** [comment]
Yeah if we are going to preserve that UI element I think it makes sense to
have a corresponding, similarly placed element. We could replace the arrow
icons with a note and a film or screen(rectangle) icon to make it a little
more intuitive.

**Comment by lucidyan** [comment]
@soya-daizu Thank you! Interesting, why not enable it by default?
